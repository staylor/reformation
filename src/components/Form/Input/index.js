import React, { Component } from 'react';
import { cx } from 'pretty-lights';
import { inputClass } from './styled';

/* eslint-disable react/prop-types */

// type Props = {
//   value: string,
//   className?: string | null,
//   onChange?: (value: string) => void,
//   bindRef?: (element: any) => void,
// };
//
// type State = {
//   value: string,
// };

export default class Input extends Component {
  state = {
    value: '',
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.value === prevState.value) {
      return null;
    }
    return { value: nextProps.value || '' };
  }

  onChange = e => {
    const value = e.target.value || '';
    this.setState({ value }, () => {
      if (this.props.onChange) {
        this.props.onChange(value);
      }
    });
  };

  render() {
    const { className, onChange, bindRef, value, ...props } = this.props;
    return (
      <input
        type="text"
        // eslint-disable-next-line
        {...props}
        ref={bindRef}
        className={cx(inputClass, className)}
        onChange={this.onChange}
        defaultValue={this.state.value}
      />
    );
  }
}

Input.defaultProps = {
  className: undefined,
  onChange: () => null,
  bindRef: () => null,
};
