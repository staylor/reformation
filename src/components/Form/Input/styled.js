import { css } from 'pretty-lights';
import { inputBaseClass } from 'components/Form/styled';
import themeUtils from 'styles/theme';

// eslint-disable-next-line
export const inputClass = css`
  ${inputBaseClass};
  display: block;
  height: 32px;
  padding: 3px 5px;
  width: 100%;

  &::placeholder {
    color: ${themeUtils.colors.detail};
  }
`;
