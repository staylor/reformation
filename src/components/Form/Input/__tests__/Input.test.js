import React from 'react';
import Input from 'components/Form/Input';
import { shallow } from 'enzyme';

const TEXT_VALUE = 'Run for the border.';

describe('Input', () => {
  test('empty', () => {
    const wrapper = shallow(<Input />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.state('value')).toEqual('');
  });

  test('add className', () => {
    const wrapper = shallow(<Input className="foo" />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('onChange', () => {
    test('adding text', () => {
      const func = jest.fn();
      const wrapper = shallow(<Input onChange={func} />);
      const value = TEXT_VALUE;
      wrapper.simulate('change', { target: { value } });

      expect(func).toHaveBeenCalledWith(value);
    });

    test('removing text', () => {
      const func = jest.fn();
      const value = TEXT_VALUE;
      const wrapper = shallow(<Input onChange={func} value={value} />);
      wrapper.simulate('change', { target: { value: '' } });

      expect(func).toHaveBeenCalledWith('');
    });

    test('bad event', () => {
      const func = jest.fn();
      const value = TEXT_VALUE;
      const wrapper = shallow(<Input onChange={func} value={value} />);
      wrapper.simulate('change', { target: { foo: 'bar' } });

      expect(func).toHaveBeenCalledWith('');
    });

    test('bad type', () => {
      const func = jest.fn();
      const value = TEXT_VALUE;
      const wrapper = shallow(<Input onChange={func} value={value} />);
      wrapper.simulate('change', { target: { value: false } });

      expect(func).toHaveBeenCalledWith('');
    });
  });
});
