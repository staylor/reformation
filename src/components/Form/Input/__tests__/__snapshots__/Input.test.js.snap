// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input add className 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  display: block;
  height: 32px;
  padding: 3px 5px;
  width: 100%;
}

.lights-0::-webkit-input-placeholder {
  color: #e6e6e6;
}

.lights-0::-moz-placeholder {
  color: #e6e6e6;
}

.lights-0:-ms-input-placeholder {
  color: #e6e6e6;
}

.lights-0::placeholder {
  color: #e6e6e6;
}

<input
  className="lights-0 foo"
  defaultValue=""
  onChange={[Function]}
  type="text"
/>
`;

exports[`Input empty 1`] = `
.lights-0 {
  background-color: #fff;
  border: 1px solid #e6e6e6;
  border-radius: 0;
  box-shadow: inset 0 1px 2px rgba(0,0,0,0.07);
  box-sizing: border-box;
  color: #32373c;
  font-size: 14px;
  outline: none;
  -webkit-transition: 0.05s border-color ease-in-out;
  transition: 0.05s border-color ease-in-out;
  display: block;
  height: 32px;
  padding: 3px 5px;
  width: 100%;
}

.lights-0::-webkit-input-placeholder {
  color: #e6e6e6;
}

.lights-0::-moz-placeholder {
  color: #e6e6e6;
}

.lights-0:-ms-input-placeholder {
  color: #e6e6e6;
}

.lights-0::placeholder {
  color: #e6e6e6;
}

<input
  className="lights-0"
  defaultValue=""
  onChange={[Function]}
  type="text"
/>
`;
